### PR TITLE
Sized check

### PR DESCRIPTION
**Pull request by tissue3**
Check if static int is the right sized int type

### CODE REVIEW

File: src/test/scala/TypeCheckerSpec.scala [hunk 1]
```diff
@@ -413,13 +413,13 @@ class TypeCheckerSpec extends FunSpec {
       typeCheck("""
               let bucket_idx = 10;
               ---
-              bucket_idx := 20;
+              bucket_idx := (20 as bit<4>);
```
Comment (rachitnigam) [inline src/test/scala/TypeCheckerSpec.scala]: You’d probably want to give a larger type to bucket_idx above.

File: src/main/scala/Parser.scala [hunk 2]
```diff
@@ -29,7 +29,8 @@ private class FuseParser extends RegexParsers with PackratParsers {
   lazy val uInt: P[Expr] = "(-)?[0-9]+".r ^^ { n => EInt(n.toInt) }
   lazy val hex = "0x[0-9a-fA-F]+".r ^^ { n => Integer.parseInt(n.substring(2), 16) }
   lazy val octal = "0[0-7]+".r ^^ { n => Integer.parseInt(n.substring(1), 8) }
-  lazy val float = "(-)?[0-9]+\\.[0-9]+".r
+  //lazy val fixed = "f'(-)?[0-9]+\\.[0-9]+".r
```
Comment (rachitnigam) [inline src/main/scala/Parser.scala]: Doesn’t need to be in this PR.

File: src/main/scala/typechecker/Subtyping.scala [hunk 3]
```diff
@@ -54,7 +55,9 @@ object Subtyping {
 
   def isSubtype(sub: Type, sup: Type): Boolean = (sub, sup) match {
     case (TSizedInt(v1, un1), TSizedInt(v2, un2)) => un1 == un2 && v1 <= v2
-    case (_:IntType, _:TSizedInt) => true
+    case (TStaticInt(v1), TSizedInt(v2,un2)) => ( (v1<0 && un2==false) | (v1>=0) ) && bitsNeeded(v1)<=v2 
+    case (_:TIndex,_:TSizedInt) => true   
+    //case (_:IntType, _:TSizedInt) => true
```
Comment (rachitnigam) [inline src/main/scala/typechecker/Subtyping.scala]: You can just delete this.

File: src/test/scala/TypeCheckerSpec.scala (hunk 1 above)
Comment (tissue3) [reply src/test/scala/TypeCheckerSpec.scala]: Yes, that's not the usual usage of as cast, but I leave it here to remind reader 10 is implicitly viewed as 4 bit number. Another example of usual usage just follows it (let test_var:bit<32> = 10;).

File: src/main/scala/typechecker/Subtyping.scala (hunk 3 above)
Comment (tissue3) [reply src/main/scala/typechecker/Subtyping.scala]: Sure wait for next PR.

**Pull request merged**